src/R/Components/Impl/History/Implementation/RHistoryEntries.cs

```diff
@@ -40,40 +49,46 @@ internal abstract class RHistoryEntries : IRHistoryEntries {
             return entry;
         }
 
-        public void Remove(IRHistoryEntry historyEntry) {
-            var entry = historyEntry as Entry;
-            if (entry == null || entry.Owner != this) {
-                throw new ArgumentException(nameof(historyEntry));
+        public void SelectRangeTo(IRHistoryEntry entry) {
+            var rangeEnd = entry as Entry;
+            if (rangeEnd == null) {
+                throw new ArgumentException(nameof(entry));
+            }
+
```

Comment:
Is this needed? if so shouldn't it be:
var rangeEnd = (Entry)entry; 
That should throw exception if entry is of type Entry.